controllers/helmchart_controller.go

```diff
@@ -447,7 +453,8 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		helmgetter.WithTimeout(repo.Spec.Timeout.Duration),
 		helmgetter.WithPassCredentialsAll(repo.Spec.PassCredentials),
 	}
-	if secret, err := r.getHelmRepositorySecret(ctx, repo); secret != nil || err != nil {
+	secret, err := r.getHelmRepositorySecret(ctx, repo)
```

Comment:
Any specific reason for this change?

Reply:
yes, we need the secret to build the getter options, but also to check if we need to login to the registry beforehand.